File: sdk/typescript/src/api/node.ts
import type { HttpClient } from "../client.js";
import type { NodeStatus, PingResponse } from "../types.js";

export class NodeApi {
  constructor(private readonly http: HttpClient) {}

  async status(): Promise<NodeStatus> {
    return this.http.get<NodeStatus>("/api/v1/status");
  }

  async ping(): Promise<PingResponse> {
    return this.http.get<PingResponse>("/api/v1/ping");
  }
}

File: sdk/cpp/src/api/node.cpp
// Copyright 2026 Hypermesh Foundation. All rights reserved.
// Licensed under the Business Source License 1.1.

#include <hypermesh/api/node.hpp>

namespace hypermesh {

NodeStatus NodeApi::status() const {
    auto j = http_.get("/api/v1/status");
    return j.get<NodeStatus>();
}

PingResponse NodeApi::ping() const {
    auto j = http_.get("/api/v1/ping");
    return j.get<PingResponse>();
}

} // namespace hypermesh
